subprojects/testfx-core/src/test/java/org/testfx/util/DebugUtilsTest.java

```diff
@@ -0,0 +1,296 @@
+
+        // then:
+        assertThat(error.getMessage(), hasAddedTextThatContains(text + other));
+    }
+
+    @Test
+    public void runCodeBlock() {
+        // given:
+        SimpleBooleanProperty prop = new SimpleBooleanProperty(false);
+
+        // when:
+        getThrownErrorPostMapper(
+            runCode(() -> prop.set(true))
+        );
+
+        // then:
+        assertThat(prop.get(), is(true));
+    }
+
+    @Test
+    public void insertHeaderIntoErrorMessage() {
+        // given:
+        String header = "header";
+
+        // when:
+        AssertionError error = getThrownErrorPostMapper(insertHeader(header, TAB));
+
+        // then:
+        assertThat(error.getMessage(), hasAddedTextEqualTo("\n\n" + TAB + header));
+    }
+
+    @Test
+    public void showPressedMouseButtonsInErrorMessage() throws TimeoutException {
+        // given:
+        Stage stage = FxToolkit.toolkitContext().getRegisteredStage();
+        moveTo(stage)
+            .moveBy(10, 10)
+            .press(MOUSE_BUTTON);
+
+        // when:
+        AssertionError error = getThrownErrorPostMapper(showMouseButtonsPressedAtTestFailure(this, TAB));
+
+        // then:
+        assertThat(error.getMessage(), hasAddedTextThatContains(MOUSE_BUTTON.name()));
+    }
+
+    @Test
+    public void showPressedKeysInErrorMessage() throws TimeoutException {
+        // given:
+        Stage stage = FxToolkit.toolkitContext().getRegisteredStage();
+        moveTo(stage)
+            .moveBy(10, 10)
+            .press(KEY);
+
+        // when:
+        AssertionError error = getThrownErrorPostMapper(showKeysPressedAtTestFailure(this, TAB));
+
+        // then:
+        assertThat(error.getMessage(), hasAddedTextThatContains(KEY.name()));
+    }
+
+    @Test
+    public void showFiredEventsInErrorMessage() throws TimeoutException {
+        // given:
+        Stage stage = FxToolkit.setupStage(stage0 -> stage0.setScene(new Scene(new Region(), 300, 300)));
+        interact(stage::show);
+        moveTo(stage).moveBy(20, 20);
+
+        // and:
+        FiredEvents events = FiredEvents.beginStoringFiredEventsOf(stage);
+        moveBy(1, 0);
+        events.stopStoringFiredEvents();
+
+        // when:
+        AssertionError error = getThrownErrorPostMapper(showFiredEvents(events, TAB));
+
+        // then:
+        assertThat(error.getMessage(), hasAddedTextThatContains("MouseEvent"));
+        assertThat(error.getMessage(), hasAddedTextNotContaining("ScrollEvent"));
+    }
+
+    private AssertionError getThrownErrorPostMapper(Function<StringBuilder, StringBuilder> errorMessageMapper) {
+        try {
+            FxAssert.verifyThat(false, alwaysFail(), errorMessageMapper);
+            throw new AssertionError("Error: verifyThat did not throw an error when it should have.");
+        }
+        catch (AssertionError error) {
+            return error;
+        }
+    }
+
+    private Matcher<Object> alwaysFail() {
+        return new AlwaysFailsMatcher();
```

Comment:
I looked into it. `IsNot` requires another Matcher to be passed into it. So, I think it's just easier to leave it as is. Besides, it also overrides the `describeMismatch()` to remove the "was" that gets appended to the description.